server/server.js

```diff
@@ -88,9 +102,13 @@ async function handleValidRequest(button, numPresses, batteryLevel) {
     }
 
     let currentSession = await db.getUnrespondedSessionWithButtonId(button.button_id, client)
+    const currentTime = await db.getCurrentTime(client)
 
     if (batteryLevel !== undefined && batteryLevel >= 0 && batteryLevel <= 100) {
-      if (currentSession === null) {
+      if (
+        currentSession === null ||
+        (currentTime - currentSession.updatedAt >= SESSION_RESET_TIMEOUT && currentSession.state === ALERT_STATE.WAITING_FOR_REPLY)
```

Comment:
Please write tests to show that this works for both the case where there is a batteryLevel and when there isn't.